### DISCUSSION TITLE
Mutating Context data in a Guard trait

### DISCUSSION THREAD

**Discussion by Dindaleon** [async-graphql/async-graphql#1189] [Q&A]
Is it possible to add or mutate data inside the context in a Guard trait? I want to check for user authorization and add the user id to the context.

I found a similar question https://github.com/async-graphql/async-graphql/discussions/934 but it is not what I want. I need to do the checking inside a Guard trait not before.

EDIT: Can you make the self parameter in the Guard trait mutable or a non-reference? That way I could send data through it instead of the context.

**Comment by sunli829** [comment]
You should to use `std::sync::Mutex` or `tokio::sync::Mutex` to lock the data.
